cpp/src/detail/shuffle_wrappers.cu

```diff
@@ -198,31 +198,73 @@ shuffle_edgelist_by_gpu_id(raft::handle_t const& handle,
                            rmm::device_uvector<int64_t>&& d_edgelist_minors,
                            std::optional<rmm::device_uvector<double>>&& d_edgelist_weights);
 
-template <typename vertex_t>
-rmm::device_uvector<vertex_t> shuffle_vertices_by_gpu_id(raft::handle_t const& handle,
-                                                         rmm::device_uvector<vertex_t>&& d_vertices)
+template <typename vertex_t, typename func_t>
+rmm::device_uvector<vertex_t> shuffle_vertices_by_gpu_id_impl(
+  raft::handle_t const& handle, rmm::device_uvector<vertex_t>&& d_vertices, func_t func)
 {
-  auto& comm           = handle.get_comms();
-  auto const comm_size = comm.get_size();
-
   rmm::device_uvector<vertex_t> d_rx_vertices(0, handle.get_stream());
   std::tie(d_rx_vertices, std::ignore) = cugraph::groupby_gpu_id_and_shuffle_values(
-    comm,  // handle.get_comms(),
+    handle.get_comms(),
     d_vertices.begin(),
     d_vertices.end(),
-    [key_func = cugraph::detail::compute_gpu_id_from_vertex_t<vertex_t>{comm_size}] __device__(
-      auto val) { return key_func(val); },
+    [key_func = func] __device__(auto val) { return key_func(val); },
     handle.get_stream());
 
   return d_rx_vertices;
 }
 
+template <typename vertex_t>
+rmm::device_uvector<vertex_t> shuffle_vertices_by_gpu_id(raft::handle_t const& handle,
```

Comment:
Should we better rename this to `shuffle_ext_vertices_by_gpu_id`?

Reply:
Done in next push